Neos.Flow/Classes/Persistence/Doctrine/Query.php

```diff
@@ -184,35 +187,35 @@ public function execute(bool $cacheResult = false): QueryResultInterface
      * This should only ever be executed from the QueryResult class.
      *
      * @return array result set
-     * @throws Exception\DatabaseException
-     * @throws Exception\DatabaseConnectionException
-     * @throws Exception\DatabaseStructureException
+     * @throws DatabaseException
+     * @throws DatabaseConnectionException
+     * @throws DatabaseStructureException
      */
     public function getResult()
     {
         try {
             $query = $this->queryBuilder->getQuery();
             if ($this->cacheResult === true || $this->settings['cacheAllQueryResults']) {
-                $query->useResultCache(true);
+                $query->enableResultCache();
             }
             return $query->getResult();
         } catch (ORMException $ormException) {
             $message = $this->throwableStorage->logThrowable($ormException);
             $this->logger->error($message, LogEnvironment::fromMethodName(__METHOD__));
             return [];
-        } catch (DBALException $dbalException) {
+        } catch (Exception $dbalException) {
             $message = $this->throwableStorage->logThrowable($dbalException);
             $this->logger->debug($message);
 
             if (stripos($dbalException->getMessage(), 'no database selected') !== false) {
```

Comment:
Note I checked this and the next "if" case and they still seem to work as intended, yes we shoudl/could find nicer ways and I would actually be in favor of just removing our custom exception wrapping and just directly throwing the original one. That said, that can be done in a separate change given that this still works fine.

Reply:
@mhsdesign FYI

Reply:
oh wow, we really checked against the exception message?! o.O

Reply:
We do 🙈

Reply:
Yes, with good reason… there is (at least was) no other way to get more details out of it.

Reply:
… and we wanted to improve DX back then, by giving a more helpful message.

Reply:
thänks a lot for investigating into this ❤️